jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -105,6 +105,9 @@ def _dataset_selected_changed(self, event={}):
             if telescope == 'JWST':
                 if 'photometry' in meta and 'pixelarea_arcsecsq' in meta['photometry']:
                     self.pixel_area = meta['photometry']['pixelarea_arcsecsq']
+                if 'bunit_data' in meta and meta['bunit_data'] == u.Unit("MJy/sr"):

```

Comment:
I think this should be inside the L106 `if` block. Because the conversion factor does not make sense if we cannot get rid of the per steradian in the first place.